### PR TITLE
fix `media.line` and `media.shunt_xxx` does not keep their network names

### LINKED ISSUE #731: inconsistent behavior when passing a network name as kwarg
Defining a name of a network sometimes works and sometimes it does not.

Two examples where passing a name works **correctly**:

Looking into the code: Media.inductor (similar: capacitor, resistor and many others) starts its definition using a match, to which the kwargs are passed. Then the match is modified to yield the result, so the name is correctly in the result.

Media.attenuator explicitly sets the name.

Examples where passing a name **does not work**:

Media.line defines a result element which includes the name (passed in kwargs). 
`result = self.match(nports=2, s_def='traveling', **kwargs)`

Then, the result element is cascaded with other elements: line uses a thru at both "ends" of the line. 
`result = self.thru(s_def='traveling')**result**self.thru(s_def='traveling')`

Network.cascade calls Network.connect, which uses a copy of the **first** element to build the result, but the thru does not carry the intended name, so the name is lost. 

Media.shunt_capacitor and shunt_inductor define a circuit element: a capacitor connected to a short, which includes the name. But note that kwargs are not passed to shunt() itself.
`return self.shunt(self.capacitor(C=C, **kwargs) ** self.short())`

Inside shunt(), the shorted capacitor is connected with an unnamed tee (because shunt() never receives any kwargs from shunt_capacitor() ). Again, the first element in connect() defines the name of the result, so the name is lost.
`return connect(self.tee(**kwargs), 1, ntwk, 0)`

Maybe it would help to include the kwargs when calling shunt(), then they will be passed to tee() and further to splitter() and match().
`return self.shunt(self.capacitor(C=C, **kwargs) ** self.short(), **kwargs)`

But I cannot oversee if that would have any unwanted side effects.

### PR DESCRIPTION
**Pull request by mhuser**
This PR attempt to fix #731 

While creating a `media.line` with `embed == True`, the name of the network, if passed into `**kwargs` was not kept due to the `media.thru()` cascading on both side.

This does not fix the faulty `embed` behaviour regarding the characteristic impedances (see #651 for reference to this issue).

`media.shunt_xxx` methods suffer from the same issue because the `**kwargs` is not passed to the `shunt` method. This PR fixes that.

### CODE REVIEW

**Review by jhillairet — APPROVED** [review]

File: skrf/media/media.py
Comment (Ttl) [inline skrf/media/media.py:842]: typo: works.

A better explanation would be that power wave reflection coefficient has conjugation. A line with characteristic impedance equal to the port impedance does not have zero reflection coefficient even with a zero length line if port impedance is complex.

File: skrf/media/media.py
Comment (Ttl) [inline skrf/media/media.py:1118]: I think `short` call also needs `**kwargs` or at least `s_def` to avoid warning of connecting different `s_def` networks when giving `s_def`.

File: skrf/media/media.py
Comment (Ttl) [inline skrf/media/media.py:1151]: Same as above.

**Review by Ttl — CHANGES_REQUESTED** [review]
Can you add tests?

File: skrf/media/media.py
Comment (mhuser) [reply skrf/media/media.py:1118]: > I think `short` call also needs `**kwargs` or at least `s_def` to avoid warning of connecting different `s_def` networks when giving `s_def`.

Indeed. I will do that.

File: skrf/media/media.py
Comment (mhuser) [reply skrf/media/media.py:842]: I will rephrase in this sense.
While this is outside the scope of this PR: isn't the `media.match` method with `npy.zeros()` wrong for the power waves ?

Comment (mhuser) [reply skrf/media/media.py]: Done

File: skrf/media/media.py
Comment (mhuser) [reply skrf/media/media.py]: Done

File: skrf/media/media.py
Comment (mhuser) [reply skrf/media/media.py]: Done

**Comment by mhuser** [comment]
> Can you add tests?

Yes, I have added test cases for the naming of networks generated by the various `media` methods right now.

**Review by jhillairet — APPROVED** [review]
Great work!

File: skrf/media/media.py
Comment (Ttl) [inline skrf/media/media.py:844]: This sounds misleading. The definition of the reflection coefficient for powerwaves has conjugation.

File: skrf/media/tests/test_media.py
Comment (Ttl) [inline skrf/media/tests/test_media.py:213]: typo

**Review by Ttl — APPROVED** [review]

File: skrf/media/media.py
Comment (mhuser) [reply skrf/media/media.py:844]: Just to be sure of my understanding: if we have a line of characteristic impedance 50 + 1j ohm, we should have port impedances of 50 - 1j ohm to have S11 equal to 0, is this correct?

Comment (mhuser) [reply skrf/media/media.py:844]: Maybe shall we just replace the 4 last comment line by ”The definition of the reflection coefficient for power waves has conjugation." 😅

File: skrf/media/tests/test_media.py
Comment (mhuser) [reply skrf/media/tests/test_media.py]: Done.

**Comment by mhuser** [comment]
@jhillairet shall we wait for the release of #733 before merging this?

**Comment by jhillairet** [comment]
> @jhillairet shall we wait for the release of #733 before merging this?

No, I think we can merge it to include it in the next release.

**Review by jhillairet — APPROVED** [review]

**Comment by mhuser** [comment]
Done !

**Pull request merged**
